source/network/network_message.h

```diff
@@ -1,23 +1,56 @@
 #pragma once
 
 #include <cinttypes>
+#include <memory>
 #include <span>
 
 #include <glm/glm.hpp>
 #include <glm/gtc/quaternion.hpp>
 
-struct message_header {
-    struct {
-        std::uint16_t size;
-    } users;
+#include "../utility/unique_span.h"
+
+struct uuid {
+    std::uint64_t values[2];
+};
+
+namespace extensions {
+    const uuid pose {0x4e166f0d6e684580, 0x865ba55945a27e83};
```

Comment:
Don't encode UUIDs as two multi-byte integers as this is endianness-specific. Use a uint8_t array instead.